src/node/history.h

```diff
@@ -175,6 +175,87 @@ namespace ccf
     {
       return crypto::Sha256Hash();
     }
+
+    std::vector<uint8_t> get_receipt(kv::Version v) override
+    {
+      return {};
+    }
+
+    bool verify_receipt(const std::vector<uint8_t>& v) override
+    {
+      return true;
+    }
+  };
+
+  class Receipt
+  {
+  private:
+    uint64_t index;
+    uint32_t max_index;
+    crypto::Sha256Hash root;
+    hash_vec* path;
+
+  public:
+    Receipt()
+    {
+      path = init_path();
+    }
+
+    static Receipt from_v(const std::vector<uint8_t>& v)
```

Comment:
Why not json or msgpack?

Reply:
Because it would buy us flexibility at the expense of execution speed and memory usage, and I don't think we need the receipt structure to be particularly flexible. Flatbuffers may be a good candidate though, since the overhead will likely be minimal.

Reply:
I think json also gives us very useful readability? A list of hashes in the `merkle_path` element of a json object is something a user can plausibly verify offline (sure they need to parse the ledger to find the tree, but that's separate from "read exactly this serialisation code"), and the `index` field should be visible to anyone else we pass this receipt to? If we pass around opaque binary blobs, the only thing a user can do with it is call `verifyRequest` on the CCF service.